pkg/registry/utils_test.go

```diff
@@ -82,9 +84,19 @@ func setup(suite *TestSuite, tlsEnabled bool, insecure bool) *registry.Registry
 	if tlsEnabled {
 		var tlsConf *tls.Config
 		tlsConf, err = tlsutil.NewClientTLS(tlsCert, tlsKey, tlsCA, insecure)
+
+		r := mockdns.Resolver{
+			Zones: map[string]mockdns.Zone{
+				"0x7f000001.": {
+					A: []string{"127.0.0.1"},
+				},
+			},
+		}
```

Comment:
Thanks for your work on this.

I tried this change in a variety of environments and it didn't always work. Along the way the default resolver is still sometimes used. :facepalm: 

I was able to make a bigger change to #11896 based on a similar idea (and same packages). After seeing it here and discussing it on the Helm dev call. See [this file](https://github.com/helm/helm/pull/11896/files#diff-6f6a553e3139587305901b6320ef20e65ea23b47a760f84e700b065e7a0239a1).